servicetalk-http-api/src/main/java/io/servicetalk/http/api/BlockingStreamingToStreamingService.java

```diff
@@ -57,6 +57,11 @@
         this.original = requireNonNull(original);
     }
 
+    @Override
+    public HttpExecutionStrategy requiredOffloads() {
+        return original.requiredOffloads();
```

Comment:
Hm... might be wrong, but it looks a bit inconsistent. Check out the ctor. It can optimize the strategy and pass the result to super class to be used for `serviceInvocationStrategy()`. The value for `requiredOffloads()` will be different. We may need to reconsider the whole approach with `serviceInvocationStrategy()` later.

Is this change required for this PR? If not, let's revert and work on it separately.

Reply:
This change is a simplification to better handle the builder `defaultStrategy()` with the API conversions and each having their own default. The builder strategy is now passed to the API conversion and the default, if appropriate, is applied in the conversion. The conversion then returns either the appropriate default strategy or the builder strategy. The builder strategy overrides the default.

This is then later combined with the filter and service strategy in the `listenForService` which allows the service and filters to adjust the strategy and add additional offloading if needed.

Reply:
Ok, can you please add some clarification comments in code to describe why `requiredOffloads()` is different from `serviceInvocationStrategy()`? 
We introduced `serviceInvocationStrategy()` in the past because `influenceStrategy` required an input strategy. With `requiredOffloads()` there is a change to remove `serviceInvocationStrategy()` in future.